### ISSUE TITLE
Overlapping frames in animated GIF and PNG

### ISSUE THREAD

**Issue by doublex** [python-pillow/Pillow#5032]
This image:
![AnimGif](https://user-images.githubusercontent.com/274624/98246874-2e18a800-1f73-11eb-8583-f2c1f7d293f0.gif)

Testcase (frame is interleaved - tested with PIL7 and PIL8):
````python
import PIL.Image
imageObject = PIL.Image.open("/tmp/98246874-2e18a800-1f73-11eb-8583-f2c1f7d293f0.gif")
imageObject.seek(5)
imageObject.seek(6)
imageObject.seek(7)
imageObject.show() 
````

**Comment by doublex** [comment]
Expected result:
![test-7](https://user-images.githubusercontent.com/274624/98355736-df2a4b80-2022-11eb-8f30-51d596daf155.png)

PIL frame:
![7](https://user-images.githubusercontent.com/274624/98355755-e3eeff80-2022-11eb-81f2-12eff691b50a.png)

**Comment by doublex** [comment]
Also APNG images won't load properly:
![ezgif-3-74c50ea21b65](https://user-images.githubusercontent.com/274624/98360169-a9d52c00-2029-11eb-9150-b6f47b4c3ca9.png)

**Comment by radarhere** [comment]
For the record, the animated PNG you have just attached shows this when the code from your first post is run.

![png](https://user-images.githubusercontent.com/3112309/98364628-3b25bc00-2085-11eb-9a77-7341af7ba28c.png)

**Comment by doublex** [comment]
Yes, there are bugs decoding AnimGIFs and APNGs (GIF is nastier).
`imagemagick` decodes the frame to:
![xx-7](https://user-images.githubusercontent.com/274624/98415633-ed598580-207d-11eb-8af5-b7e45860d2f9.png)

**Comment by stevemcgf** [comment]
I only see the issue with APNG files not with GIF.

**Comment by stevemcgf** [comment]
Animated PNG
![sticker02](https://user-images.githubusercontent.com/878889/100572064-75037000-3289-11eb-8df1-35592ee8e7ad.png)

Incorrect render of 2nd frame
![tmpmvjf3dp4](https://user-images.githubusercontent.com/878889/100572084-80569b80-3289-11eb-99dc-5cc83fd3ab49.PNG)

Fixed render of 2nd frame
![tmp1l63_0g8](https://user-images.githubusercontent.com/878889/100572125-98c6b600-3289-11eb-916e-392d4da6175f.PNG)

**Comment by doublex** [comment]
@stevemcgf 

AnimGIF renders wrong (at least for me using pillow 8.0.1).
Steps to reproduce:
````
wget https://user-images.githubusercontent.com/274624/98246874-2e18a800-1f73-11eb-8583-f2c1f7d293f0.gif
````


````python
import PIL.Image
image = PIL.Image.open("/tmp/98246874-2e18a800-1f73-11eb-8583-f2c1f7d293f0.gif")
for frame in PIL.ImageSequence.Iterator(image):
    last_frame = frame
last_frame.show()
````

**Comment by radarhere** [comment]
The GIF part of this should be resolved by #5125

**Comment by radarhere** [comment]
I've created PR #5126 to resolve the APNG part of this, as an alternative to #5071

**Comment by radarhere** [comment]
The GIF part of this should be fixed in Pillow 8.1.0, due to be released on January 2.

**Comment by doublex** [comment]
Thanks for the effort!

**Comment by radarhere** [comment]
The APNG part of this should now also be fixed in Pillow 8.1.0

**Comment by doublex** [comment]
@radarhere 
Thanks again for your efforts!
